roles/apply_nmstate/tasks/main.yml

```diff
@@ -3,23 +3,36 @@
   become: true
   block:
     - name: Install nmstate
-      package:
+      ansible.builtin.package:
         name: nmstate
         state: present
 
     - name: "Copy rendered_nmstate_yml to {{ vm_nmstate_config_path }}"
-      copy:
+      ansible.builtin.copy:
         content: "{{ rendered_nmstate_yml }}"
         dest: "{{ vm_nmstate_config_path }}"
         mode: 0644
       # No commit is done to revert the changes if they cause the host to be come unreachable
 
+    - name: Check if vm_host is bastion
+      ansible.builtin.include_tasks:
+        file: check_if_vm_host_is_bastion.yml
+
     - name: Apply nmstate
-      shell:
+      ansible.builtin.shell:
         cmd: "nmstatectl apply --no-commit --timeout 120 {{ vm_nmstate_config_path }}"
       async: 60
       poll: 5
 
+    - name: "Check for connection wider network"
+      ansible.builtin.shell:
+        cmd: "ping -c 4 {{ vm_network_test_ip }}"
```

Comment:
This ping will default from the interface with the default route. Also, I believe the ping default timeout is 4secs. You might want to bring the timeout down to 1sec.

Reply:
I think default route is fine as it's just to check we still can connect in if need be. I've added the 1s timeout though